### PR TITLE
Add copyFile to node

### PR DESCRIPTION
**Pull request by Loghorn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v8.x/docs/api/fs.html#fs_fs_copyfile_src_dest_flags_callback
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -4285,6 +4288,54 @@ declare module "fs" {
...
+    /**
+     * Asynchronously copies src to dest. By default, dest is overwritten if it already exists.
+     * No arguments other than a possible exception are given to the callback function.
+     * Node.js makes no guarantees about the atomicity of the copy operation.
+     * If an error occurs after the destination file has been opened for writing, Node.js will attempt
+     * to remove the destination.
+     * @param src A path to the source file.
+     * @param dest A path to the destination file.
+     */
+    export function copyFile(src: PathLike, dest: PathLike, callback: (err: NodeJS.ErrnoException) => void): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think it should be `NodeJS.ErrnoException | null`.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -4285,6 +4288,54 @@ declare module "fs" {
...
+     * Asynchronously copies src to dest. By default, dest is overwritten if it already exists.
+     * No arguments other than a possible exception are given to the callback function.
+     * Node.js makes no guarantees about the atomicity of the copy operation.
+     * If an error occurs after the destination file has been opened for writing, Node.js will attempt
+     * to remove the destination.
+     * @param src A path to the source file.
+     * @param dest A path to the destination file.
+     * @param flags An integer that specifies the behavior of the copy operation. The only supported flag is fs.constants.COPYFILE_EXCL, which causes the copy operation to fail if dest already exists.
+     */
+    export function copyFile(src: PathLike, dest: PathLike, flags: number, callback: (err: NodeJS.ErrnoException) => void): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think it should be `NodeJS.ErrnoException | null`.

Comment (Loghorn) [reply types/node/index.d.ts]: I think you are right, but all the other callbacks are declared this way so I kept the same style

Comment (Flarna) [reply types/node/index.d.ts]: ok fine with me. Maybe it's better to change them all via a separated PR.

**Comment by Flarna** [comment]
LGTM for changes in node.

**Review by vvakame — APPROVED** [review]

**Comment by andy-ms** [comment]
@Loghorn Could you remove unrelated changes? If there are test failures in other packages they should be changed in other PRs.

**Comment by Loghorn** [comment]
@andy-ms Done

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**